### ISSUE TITLE
Cromwell needs a way to scatter over maps

### ISSUE THREAD

**Issue by yfarjoun** [broadinstitute/cromwell#2261]
AC: 

Can scatter over a map where the scatter element is the Pair[key, value]

**Comment by cjllanwarne** [comment]
How would this be as a temporary enabler (I think this will be easier/faster than modifying the scatter logic itself):

```
Map[X,Y] map # from elsewhere
List[Pair[X,Y]] listOfPair = map # Coerce to a list of pairs
scatter (pair in listOfPair) {
  # do the scatter
}
```

**Comment by yfarjoun** [comment]
If the coercion works, that would be fine. 

Bonus points for making the coersion work implicitly...

**Comment by cjllanwarne** [comment]
Resolved by #2274.
